packages/common-integration-tests/src/token-scope.ts

```diff
@@ -0,0 +1,73 @@
+import {IAuthClient} from '@gomomento/sdk-core/dist/src/clients/IAuthClient';
+import {ICacheClient} from '@gomomento/sdk-core/dist/src/clients/ICacheClient';
+import {
+  AllDataReadWrite,
+  CacheGet,
+  CacheSet,
+  CreateCache,
+  DeleteCache,
+  ExpiresIn,
+  GenerateAuthToken,
+  MomentoErrorCode,
+} from '@gomomento/sdk-core';
+
+export function runTokenScopeTests(
+  v1SuperUserToken: string,
+  authClientFactory: (token: string) => IAuthClient,
+  cacheClientFactory: (token: string) => ICacheClient,
+  cacheName: string
+) {
+  describe('AllDataReadWrite token scope', () => {
+    let allDataReadWriteClient: ICacheClient;
+
+    beforeAll(async () => {
+      const authClient = authClientFactory(v1SuperUserToken);
+      const generateResponse = await authClient.generateAuthToken(
+        AllDataReadWrite,
+        ExpiresIn.seconds(60)
+      );
+      expect(generateResponse).toBeInstanceOf(GenerateAuthToken.Success);
+      const allDataReadWriteToken = (
+        generateResponse as GenerateAuthToken.Success
+      ).authToken;
+      allDataReadWriteClient = cacheClientFactory(allDataReadWriteToken);
+    });
+    it('cannot create a cache', async () => {
+      const createCacheResponse = await allDataReadWriteClient.createCache(
+        'FOOFOOFOO'
+      );
+      expect(createCacheResponse).toBeInstanceOf(CreateCache.Error);
+      const createCacheError = createCacheResponse as CreateCache.Error;
+      expect(createCacheError.errorCode()).toEqual(
+        MomentoErrorCode.AUTHENTICATION_ERROR
```

Comment:
I just merged the change that changes the grpc status from `UNAUTHENTICATED` to `PERMISSION_DENIED`

Reply:
okay well we can fix that on a subsequent PR when we see it fail?

Reply:
sure I just think it might fail as soon as you merge this in to main 😅